crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -2660,6 +2660,51 @@ impl<'db> TypeInferenceBuilder<'db> {
                 self.infer_binary_type_comparison(left, op, KnownClass::Str.to_instance(self.db))
             }
 
+            (Type::BytesLiteral(salsa_b1), Type::BytesLiteral(salsa_b2)) => {
+                let contains_subsequence = |needle: &[u8], haystack: &[u8]| {
```

Comment:
Nit: We could consider using https://docs.rs/memchr/latest/memchr/memmem/fn.find.html

Reply:
Yes, thanks. Did a grep for `memmem` , didn't find anything, and wasn't sure if it warranted adding a new dependency. Should have looked for `memchr`.

==> #13750